packages/disputer/test/Disputer.js

```diff
@@ -576,89 +571,364 @@ contract("Disputer.js", function(accounts) {
+              priceFeedMock.setLastUpdateTime(earliestLiquidationTime + 2);
+              priceFeedMock.setLookback(1);
+              await disputer.update();
+              await disputer.dispute(convertPrice("1.0"));
+              assert.equal(spy.callCount, 1); // 1 info level logs should be sent for the dispute
+              assert.equal((await emp.getLiquidations(sponsor1))[0].state, LiquidationStatesEnum.PENDING_DISPUTE);
+
+              // The disputeBot should be the disputer in sponsor1  liquidations.
+              assert.equal((await emp.getLiquidations(sponsor1))[0].disputer, disputeBot);
+            }
           );
-
-          // Say the price feed reports a price of 1 USD per token. This makes the liquidation invalid and the disputer should
-          // dispute the liquidation: 125/(100*1.0)=1.25 CR -> Position was collateralized and invalid liquidation.
-          priceFeedMock.setHistoricalPrice(convertPrice("1"));
-
-          // However, say disputer operator has provided an override price of 1.2 USD per token. This makes the liquidation
-          // valid and the disputer should do nothing: 125/(100*1.2)=1.0
-          await disputer.update();
-          const earliestLiquidationTime = Number(empClient.getUndisputedLiquidations()[0].liquidationTime);
-          priceFeedMock.setLastUpdateTime(earliestLiquidationTime);
-          await disputer.dispute(convertPrice("1.2"));
-          assert.equal(spy.callCount, 0); // 0 info level logs should be sent as no dispute.
-          assert.equal((await emp.getLiquidations(sponsor1))[0].state, LiquidationStatesEnum.PRE_DISPUTE);
-
-          // Next assume that the override price is in fact 1 USD per token. At this price point the liquidation is now
-          // invalid that the disputer should try dispute the tx. This should work even if the liquidation timestamp is
-          // earlier than the price feed's earliest available timestamp:
-          priceFeedMock.setLastUpdateTime(earliestLiquidationTime + 2);
-          priceFeedMock.setLookback(1);
-          await disputer.update();
-          await disputer.dispute(convertPrice("1.0"));
-          assert.equal(spy.callCount, 1); // 1 info level logs should be sent for the dispute
-          assert.equal((await emp.getLiquidations(sponsor1))[0].state, LiquidationStatesEnum.PENDING_DISPUTE);
-
-          // The disputeBot should be the disputer in sponsor1  liquidations.
-          assert.equal((await emp.getLiquidations(sponsor1))[0].disputer, disputeBot);
+          describe("disputer correctly deals with funding rates from perpetual contract", () => {
+            versionedIt([{ contractType: "Perpetual", contractVersion: "latest" }])(
+              "Can correctly detect invalid liquidations and dispute them",
+              async function() {
+                // sponsor1 creates a position with 125 units of collateral, creating 100 synthetic tokens.
+                await emp.create(
```

Comment:
Shouldn't this be `perpetual` instead of `emp` throughout? Is the outdated name scheme for convenience, due to some set-up of `emp` in the tests?

Reply:
for now I've left the `emp` syntax as this is through the repo. the churn for this change will be very large. we also have things like `ExpiriningMultiPartyClient` that would need to be renamed.

In a follow-up PR once all bots are updated, I will do a massive find and replace to update this syntax through the bots.